### PR TITLE
Feature/add abstract field to complex search

### PR DESCRIPTION
**Pull request by DominikVoigt**
This PR adds the abstract field as a fielded term for complex queries and implements it for the existing complex search fetchers that support abstract field based filtering.

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java [hunk 1]
```diff
@@ -118,6 +128,7 @@ public int hashCode() {
         int result = defaultField != null ? defaultField.hashCode() : 0;
         result = 31 * result + (getAuthors() != null ? getAuthors().hashCode() : 0);
         result = 31 * result + (getTitlePhrases() != null ? getTitlePhrases().hashCode() : 0);
+        result = 31 * result + (getAbstractPhrases() != null ? getTitlePhrases().hashCode() : 0);
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java:131]: Why not `return Objects.hash(defaultField, getAuthors(), ...)`?

File: src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java [hunk 2]
```diff
@@ -134,14 +145,16 @@ public String toString() {
         getJournal().ifPresent(journal -> stringRepresentation.append(journal).append(" "));
         stringRepresentation.append(String.join(" ", getTitlePhrases()))
                             .append(String.join(" ", getDefaultFieldPhrases()))
-                            .append(String.join(" ", getAuthors()));
+                            .append(String.join(" ", getAuthors()))
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/fetcher/ComplexSearchQuery.java:148]: While you are on it, please switch to [StringJoiner](https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html)

```java
StringJoiner stringJoiner = new StringJoiner(" ");
getSingleYear().ifPresent(singleYear -> stringJoiner.append(singleYear));
...
```

**Review by koppor — CHANGES_REQUESTED** [review]
Some small nitpick comment

**Review by koppor — APPROVED** [review]

**Review by calixtus — APPROVED** [review]
LGTM

**Comment by calixtus** [comment]
Two approvals -> I think I'll merge this.
Thanks for the work. I hope you succeeded in your exams well.

**Pull request merged**